Remotion/Core/Collections.Caching.UnitTests/LazyLockingCachingAdapterTest.cs

```diff
@@ -56,21 +57,29 @@ public void GetOrCreateValue_TryGetValueIsFalse ()
     {
       var value = new object ();
       var doubleCheckedLockingContainer = CreateContainerThatChecksForNotProtected (value);
+      Lazy<Wrapper> outResult = null;
 
       _innerCacheMock
-          .Expect (mock => mock.TryGetValue (Arg.Is ("key"), out Arg<Lazy<Wrapper>>.Out (null).Dummy))
-          .Return (false)
-          .WhenCalled (mi => CheckInnerCacheIsProtected ());
+          .Setup (mock => mock.TryGetValue ("key", out outResult))
+          .Returns (false)
+          .Callback (
+              (OutDelegate) ((string key, out Lazy<Wrapper> value) =>
+              {
+                value = doubleCheckedLockingContainer;
```

Comment:
The diff doesn't really show where this line comes from. What am I missing?

Reply:
`value` must be assigned inside the `Callback` block, it has no other purpose

Reply:
should we then not use `null`? By using an exisiting value, a semantic is implied.

Reply:
I would suggest the following to keep things readable:
```csharp
_innerCacheMock
    .Setup (mock => mock.TryGetValue ("key", out outResult))
    .Returns (false)
    .Callback (new InvocationAction (_ => CheckInnerCacheIsProtected()))
    .Verifiable();
```